packages/amplify-cli/src/extensions/amplify-helpers/resource-status-diff.ts

```diff
@@ -0,0 +1,295 @@
+    cloudTemplate : {
+                        [key: string]: any;
+                    };
+
+    constructor( category, resourceName, provider ){
+        this.localBackendDir = pathManager.getBackendDirPath();
+        this.cloudBackendDir = pathManager.getCurrentCloudBackendDirPath();
+        this.resourceName = resourceName;
+        this.category = category;
+        this.provider = this.normalizeProviderForFileNames(provider);
+        this.service = getResourceService(category, resourceName);
+        this.localTemplate = {}; //requires file-access, hence loaded from async methods
+        this.cloudTemplate = {}; //requires file-access, hence loaded from async methods
+        //Note: All file names include full-path but no extension.Extension will be added later.
+        this.resourceFiles = {
+            localTemplateFile : path.normalize(path.join(this.localBackendDir, category, resourceName)),
+            cloudTemplateFile : path.normalize(path.join(this.cloudBackendDir, category, resourceName)),
+            //Paths using glob for Cfn file and Build file
+            localPreBuildCfnFile : this.globCfnFilePath( path.normalize(path.join(this.localBackendDir, category, resourceName))),
+            cloudPreBuildCfnFile : this.globCfnFilePath( path.normalize(path.join(this.cloudBackendDir, category, resourceName))),
+            localBuildCfnFile : this.globCfnFilePath( path.normalize(path.join(this.localBackendDir, category, resourceName, 'build'))),
+            cloudBuildCfnFile : this.globCfnFilePath( path.normalize(path.join(this.cloudBackendDir, category, resourceName, 'build'))),
+          }
+    }
+
+    //API :View: Print the resource detail status for the given mutation (Create/Update/Delete)
+    public printResourceDetailStatus = async ( mutationInfo : StackMutationInfo ) => {
+      const header = `${ mutationInfo.consoleStyle(mutationInfo.label)}`;
+      const diff = await this.calculateCfnDiff();
+      print.info(`${resourceDetailSectionStyle(`[\u27A5] Resource Stack: ${capitalize(this.category)}/${this.resourceName}`)} : ${header}`);
+      const diffCount = this.printStackDiff( diff );
+      if ( diffCount === 0 ){
+          console.log("No changes  ")
+      }
+    }
+
+    //API :Data: Calculate the difference in cloudformation templates between local and cloud templates
+    public calculateCfnDiff = async () => {
+      const resourceTemplatePaths = await this.getCfnResourceFilePaths();
+      //load resource template objects
+      this.localTemplate = await this.loadCfnTemplate(resourceTemplatePaths.localTemplatePath)
+      this.cloudTemplate = await this.loadCfnTemplate(resourceTemplatePaths.cloudTemplatePath);
+      const diff : cfnDiff.TemplateDiff = cfnDiff.diffTemplate(this.cloudTemplate, this.localTemplate );
+      return diff;
+    }
+
+    //helper: Select cloudformation file path from build folder or non build folder.
+    private getCfnResourceFilePaths = async() => {
+      return {
```

Comment:
I think we have to return a promise here
return new Promise({}) for async/await fn

Reply:
The function is defined as async. This automatically wraps the return value into a promise.